riscv-page-tables/src/page_table.rs

```diff
@@ -570,12 +609,12 @@ impl<T: PagingMode> PageTableInner<T> {
             }
             Locked(_) => Err(Error::PteLocked),
             Leaf(_) => Err(Error::MappingExists),
-            Table(_) => unreachable!(),
+            Table(_) => unreachable!(), // good test of problems with page sizes
```

Comment:
This is case is no longer unreachable in the presence of huge pages. Consider the case where a range of address space is already mapped with 4kB pages and the user tries to map in 2MB pages. We should return an error instead of blowing up here.

Reply:
fixed.